spec/capybara_helper.rb

```diff
@@ -10,3 +25,86 @@ def set_implicit_central!
   @request.try { |it| it.host = Mumukit::Platform.laboratory.domain }
   Capybara.app_host = Mumukit::Platform.laboratory.url
 end
+
+# Config helpers
+
+def selected_driver
+  case ENV['MUMUKI_SELENIUM_DRIVER']
+  when 'chrome'
+    :selenium_chrome_headless
+  when 'firefox'
+    :selenium_headless
+  when 'safari'
+    :selenium_safari
+  else
+    :rack_test
+  end
+end
+
+def run_with_selenium?
+  selected_driver.to_s.start_with? 'selenium'
+end
+
+def register_safari_driver!
+  # No official docs for this, code was taken from https://github.com/teamcapybara/capybara/blob/master/spec/selenium_spec_safari.rb
+  Capybara.register_driver :selenium_safari do |app|
+    Capybara::Selenium::Driver.new(app, browser: :safari, timeout: 30)
+  end
+end
+
+# Registers a middleware to set request headers, because Selenium Webdriver has no native support for that
+def register_request_headers_workaround!
+  RSpec.configure do |config|
+    config.after(:suite) do
+      $custom_headers = nil
+    end
+  end
+
+  Object.class_eval <<-EOR
+    module RequestWithExtraHeaders
+      def headers
+        $custom_headers.each { |key, value| self.set_header "HTTP_\#{key}", value } if $custom_headers
+        super
+      end
+    end
+
+    class ActionDispatch::Request
+      prepend RequestWithExtraHeaders
+    end
+  EOR
+end
+
+def exclude_selenium_failing_tests!
+  RSpec.configure do |config|
+    config.filter_run_excluding(
```

Comment:
Does rspec when these tests are ignored?

Reply:
By default, it outputs the following info when `filter_run_excluding` is used:

```
Run options: exclude {:xpath_no_matches=>true, :organization_not_nil=>true, :navigation_error=>true, :json_eq_error=>true, :invalid_selector_error=>true, :element_not_interactable_error=>true, :http_response_headers=>true}
```

On top of that, I've added another line to be sure it's running with the desired Selenium driver:

```
Running Capybara tests with selenium_headless, not ignoring hidden elements`
```